include/envoy/network/transport_socket.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/common/pure.h"
+#include "envoy/network/connection.h"
+
+namespace Envoy {
+namespace Network {
+
+/**
+ * Action that should occur on a connection after I/O.
+ */
+enum class PostIoAction {
+  // Close the connection.
+  Close,
+  // Keep the connection open.
+  KeepOpen
+};
+
+/**
+ * Result of each I/O event.
+ */
+struct IoResult {
+  PostIoAction action_;
+
+  /**
+   * Number of bytes processed by the I/O event.
+   */
+  uint64_t bytes_processed_;
+};
+
+/**
+ * Callbacks used by transport socket instances to communicate with connection
+ */
+class TransportSocketCallbacks {
+public:
+  virtual ~TransportSocketCallbacks() {}
+
+  /**
+   * @return the file descriptor associated with the connection.
```

Comment:
We're not terribly consistent about it, but new Envoy style has the return type specified after @return
This should be applied to all @returns in this file

Reply:
Also mild preference for Connection or Network::Connection where we're actually talking about a Network::Connection

Reply:
Done